donate/payments/tasks.py

```diff
@@ -75,11 +75,28 @@ def process_subscription_charged_successfully(self, notification):
         customer = gateway.customer.find(payment_method.customer_id)
         custom_fields = customer.custom_fields or {}
 
+        # The details of a donor are in a different spot depending on the payment method
+
+        if is_paypal:
+            donor_details = {
+                "last_name": last_tx.paypal_details.payer_last_name,
+                "first_name": last_tx.paypal_details.payer_first_name,
+                "email": last_tx.paypal_details.payer_email
+            }
+        elif last_tx.payment_instrument_type == 'credit_card':
+            donor_details = {
+                'last_name': last_tx.customer_details.last_name,
+                'first_name': last_tx.customer_details.first_name,
+                'email': last_tx.customer_details.email,
+            }
+        else:
+            logger.error("Unexpected payment type on subscription webhook")
+
         send_transaction_to_basket({
-            'last_name': last_tx.customer_details.last_name,
-            'first_name': last_tx.customer_details.first_name,
+            'last_name': donor_details.get('last_name', ''),
+            'first_name': donor_details.get('first_name', ''),
             'campaign_id': custom_fields.get('campaign_id', ''),
-            'email': last_tx.customer_details.email,
+            'email': donor_details.get('email', ''),
```

Comment:
Curious if the use of `.get()` with a default of `''` is necessary? are we expecting these keys to sometimes be empty?

Reply:
If email is empty, basket will barf. But I suppose name could be empty at some point. (And we can't use JS-like object notation because it's a dict, not a proper object. I found that out when tests were failing.)

Reply:
But I don't think we are expecting email to ever be empty in this case.

Reply:
Well the alternative to `foo.get('bar')` is `foo['bar']` for dicts - which is why I bring it up, since we can just do that if we don't expect to have to set default values.

Reply:
I could have sworn I'd tried that and it wasn't happy.

Reply:
the problem with `foo['bar']` is if `bar` doesn't exist in `foo`, it will throw an exception, whereas `foo.get('bar')` will return `None` without throwing.

Reply:
It's fine with me either way. If we don't trust that Braintree is going to give us consistent data in webhooks we can use `.get()`